ogre/src/OgreHeightmap.cc

```diff
@@ -0,0 +1,3304 @@
+        _v[tileIndex].push_back(_v[tileIndex].back());
+
+        tileIndex = tileR * sqrt(_n) +
+            (tileIndex + 1) % static_cast<int>(sqrt(_n));
+      }
+      ++count;
+    }
+    // Copy the last row
+    for (int i = 0; i < sqrt(_n); ++i)
+    {
+      tileIndex = tileR * sqrt(_n) + i;
+      std::vector<float> lastRow(_v[tileIndex].end() - newWidth,
+          _v[tileIndex].end());
+      _v[tileIndex].insert(_v[tileIndex].end(),
+          lastRow.begin(), lastRow.end());
+    }
+  }
+}
+
+/////////////////////////////////////////////////
+void OgreHeightmap::DefineTerrain(const int _x, const int _y)
+{
+  Ogre::String filename = this->dataPtr->terrainGroup->generateFilename(_x, _y);
+
+  bool resourceExists =
+      Ogre::ResourceGroupManager::getSingleton().resourceExists(
+      this->dataPtr->terrainGroup->getResourceGroup(), filename);
+
+  if (resourceExists && !this->dataPtr->terrainHashChanged)
+  {
+    ignmsg << "Loading heightmap cache data: " << filename << std::endl;
+
+    this->dataPtr->terrainGroup->defineTerrain(_x, _y);
+    this->dataPtr->loadedFromCache = true;
+  }
+  else
+  {
+    if (this->dataPtr->splitTerrain)
+    {
+      // generate the subterrains if needed
+      if (this->dataPtr->subTerrains.empty())
+      {
+        this->SplitHeights(this->dataPtr->heights,
+            this->dataPtr->numTerrainSubdivisions,
+            this->dataPtr->subTerrains);
+      }
+
+      this->dataPtr->terrainGroup->defineTerrain(_x, _y,
+          &this->dataPtr->subTerrains[this->dataPtr->terrainIdx][0]);
+      ++this->dataPtr->terrainIdx;
+    }
+    else
+    {
+      this->dataPtr->terrainGroup->defineTerrain(_x, _y,
+          &this->dataPtr->heights[0]);
+    }
+  }
+}
+
+/////////////////////////////////////////////////
+void OgreHeightmap::CreateMaterial()
+{
+  if (!this->dataPtr->materialName.empty())
+  {
+    // init custom material generator
+    Ogre::TerrainMaterialGeneratorPtr terrainMaterialGenerator;
+    TerrainMaterial *terrainMaterial = OGRE_NEW TerrainMaterial(
+        this->dataPtr->materialName);
+    if (this->dataPtr->splitTerrain)
+      terrainMaterial->setGridSize(this->dataPtr->numTerrainSubdivisions);
+    terrainMaterialGenerator.bind(terrainMaterial);
+    this->dataPtr->terrainGlobals->setDefaultMaterialGenerator(
+        terrainMaterialGenerator);
+  }
+  else
+  {
+#if OGRE_VERSION_MAJOR == 1 && OGRE_VERSION_MINOR < 11
+    // use default material
+    // RTSS PSSM shadows compatible terrain material
+    if (!this->dataPtr->ignMatGen)
+      this->dataPtr->ignMatGen = new IgnTerrainMatGen();
+
+    auto ptr = Ogre::TerrainMaterialGeneratorPtr();
+    ptr.bind(this->dataPtr->ignMatGen);
+
+    this->dataPtr->terrainGlobals->setDefaultMaterialGenerator(ptr);
+#else
+    // init custom material generator
+    Ogre::TerrainMaterialGeneratorPtr terrainMaterialGenerator;
+    auto terrainMaterial = OGRE_NEW TerrainMaterial("Ignition/Grey");
```

Comment:
`Ignition/Grey` does not exist. You can use [Default/White](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering4/src/base/BaseScene.cc#L1275) since this should just be a dummy material.